lib/plugins/blocks.js

```diff
@@ -104,16 +104,35 @@ function inject (bot, { version, storageBuilder }) {
   }
 
   async function waitForChunksToLoad () {
-    // check 5x5 chunks around us
-    for (let x = -2; x <= 2; x++) {
-      for (let y = -2; y <= 2; y++) {
-        if (bot.world.getColumnAt(bot.entity.position.plus(new Vec3(x, 0, y).scaled(16))) == null) {
-          // keep wait
-          await sleep(100)
-          await waitForChunksToLoad()
+    return new Promise((resolve) => {
+      // get corner coords of 5x5 chunks around us
+      let chunkPosToCheck = []
+      for (let x = -2; x <= 2; x++) {
+        for (let y = -2; y <= 2; y++) {
+          chunkPosToCheck.push(bot.entity.position.plus(new Vec3(x, 0, y).scaled(16)))
         }
       }
-    }
+
+      // ignore any chunks which are already loaded
+      chunkPosToCheck = chunkPosToCheck.filter((vectorElement) => { return bot.world.getColumnAt(vectorElement) == null })
+
+      function waitForLoadEvents (columnCorner) {
+        const position = chunkPosToCheck.findIndex((vectorElement) => { return columnCorner.equals(vectorElement) })
+        if (position !== -1) { // loaded chunk was a desired chunk, no longer need to check it
+          chunkPosToCheck.splice(position, 1)
+        }
+        if (chunkPosToCheck.length === 0) { // no chunks left to find
+          this.bot.world.off('chunkColumnLoad', waitForLoadEvents) // remove this listener instance
+          resolve()
+        }
+      }
+      
+      if (chunkPosToCheck.length > 0) { // begin listening for remaining chunks to load
+        this.bot.world.on('chunkColumnLoad', waitForLoadEvents)
+      } else { // all chunks were already loaded, skip listening for events
+        resolve()
+      }
```

Comment:
This could be simplified using the `once` helper and `Promise.all()`

Reply:
I might be misunderstanding your comment, but there are multiple chunks being loaded. If the code is run as a once, to the best of my knowledge there's no easy way to keep listening for other chunks if this one chunk isn't the one we want. If there is, then just running a Promise.all() on the map of a filtered list of missing chunks would be nice.

Reply:
Yes you are right. My comment was a bit quick. However adding an event that include the chunk coordinates could be useful: https://github.com/PrismarineJS/prismarine-world/blob/master/src/world.js#L109

Reply:
yeah agreed, this event is missing

Reply:
https://github.com/PrismarineJS/prismarine-world/blob/master/docs/API.md#blockupdatex-y-z-oldblock-newblock we already have one for blocks, having one for chunks would be coherent